lib/models/wallet_connect/wallet_connect_modals_controller.dart

```diff
@@ -64,7 +64,45 @@ class WalletConnectModalsController {
     }
   }
 
-  //Handles sending Qubic
+  Future<ApproveSendTransactionResult> handleSendTransaction(
+      RequestSendTransactionEvent event, BuildContext context) async {
+    await _autoIgnoreRequestsWhenModalIsOpen(event.topic, event.requestId);
+    _wCDialogOpen = true;
+
+    try {
+      var result = await Navigator.of(context)
+          .push(MaterialPageRoute<ApproveSendTransactionResult>(
+              builder: (BuildContext context) {
+                return ApproveTokenTransfer(
+                    pairingMetadata: event.pairingMetadata!,
+                    fromID: event.fromID,
+                    fromName: event.fromIDName,
+                    amount: event.amount,
+                    toID: event.toID,
+                    inputType: event.inputType,
+                    payload: event.payload);
+              },
+              fullscreenDialog: true));
+      _wCDialogOpen = false;
+      if (result == null) {
+        var err = Errors.getSdkError(Errors.USER_REJECTED);
+        throw JsonRpcError(code: err.code, message: err.message);
+      } else {
+        if ((result.errorCode == null) && (result.errorMessage == null)) {
+          return result;
+        } else {
+          throw JsonRpcError(
+              code: result.errorCode ?? -1,
```

Comment:
@ahmed-tarek-salem what his -2 means? if we have a list of error codes, we should define as constant and document for the DApps

Reply:
to be solve later